server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -6165,6 +6165,28 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
         return vm;
     }
 
+    private boolean checkSecurityGroupSupportForNetwork(DataCenter zone, List<Long> networkIds,
+                                                        List<Long> securityGroupsIds) {
+        if (zone.isSecurityGroupEnabled()) {
+            return true;
+        }
+        if (CollectionUtils.isNotEmpty(networkIds)) {
+            for (Long networkId : networkIds) {
+                Network network = _networkDao.findById(networkId);
+                if (network == null) {
+                    throw new InvalidParameterValueException("Unable to find network by id " + networkId);
+                }
+                if (network.getGuestType() == Network.GuestType.Shared && _networkModel.isSecurityGroupSupportedInNetwork(network)) {
+                    return true;
+                }
+            }
+        } else if (CollectionUtils.isNotEmpty(securityGroupsIds)) {
+            Network networkWithSecurityGroup = _networkModel.getNetworkWithSGWithFreeIPs(zone.getId());
```

Comment:
@vishesh92 
can you explain why added this ?

Reply:
If security group IDs are not empty, then we use a default security group in `createAdvancedSecurityGroupVirtualMachine`.

Reply:
got it

as discussed, if securityGroupsIds is specified but networkids are not, ACS will choose a shared network with SG to deploy the vm